mmdeploy/codebase/mmflow/models/decoders/raft_decoder.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+from typing import Dict, Optional, Sequence, Union
+
+import numpy as np
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmflow.models.decoders.raft_decoder.CorrelationPyramid.forward',backend='default')
+def raft_decoders_forward(ctx,self, feat1: torch.Tensor,
+            feat2: torch.Tensor) -> Sequence[torch.Tensor]:
+    """Forward function for Correlation pyramid.
+
+    Args:
+        feat1 (Tensor): The feature from first input image.
+        feat2 (Tensor): The feature from second input image.
+
+    Returns:
+        Sequence[Tensor]: The list of correlation which is pooled using
+            average pooling with kernel sizes {1, 2, 4, 8}.
+    """
+    N, C, H, W = feat1.shape
+    corr = torch.matmul(
+        feat1.view(N, C, -1).permute(0, 2, 1),
+        feat2.view(N, C, -1)).view(N, H, W, H, W)
+    corr = corr.reshape(N * H * W, 1, H, W) / torch.sqrt(C.clone().detach().float())
```

Comment:
It is not safe to use clone here. C could be int type if dynamic_axes is None when export onnx.